### PR TITLE
updates to readme and changelog

### PR DESCRIPTION
**Pull request by LisaDelaney**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,168 +1,249 @@
...
+* Extended benchmark suite of `DeviceHistogram`, `DeviceScan`, `DevicePartition`, `DeviceReduce`,
+`DeviceSegmentedReduce`, `DeviceSegmentedRadixSort`, `DeviceRadixSort`, `DeviceSpmv`,
+`DeviceMergeSort`, and `DeviceSegmentedSort`
+
+### Changes
+
+* Obsolete type traits defined in `util_type.hpp`; use the standard library equivalents instead
+* CUB backend references CUB and Thrust version 1.16.0
+* `DeviceRadixSort` `num_items` parameter type is now templated instead of being an int
+  * If an integral type with a maximum size of 4 bytes is passed (an int), the former logic applies;
```
Comment (LisaDelaney) [inline CHANGELOG.md:109]: Does this bullet point look correct? (e.g., "maximum size of 4 bytes" and "2**32 elements")

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,168 +1,249 @@
...
+    otherwise, the algorithm uses a larger indexing type that makes it possible to sort input data over
+    2**32 elements
```
Comment (saadrahim) [inline CHANGELOG.md]:
```suggestion
    $2^32$ elements
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,168 +1,249 @@
...
+`DeviceMergeSort`, and `DeviceSegmentedSort`
+
+### Changes
+
+* Obsolete type traits defined in `util_type.hpp`; use the standard library equivalents instead
+* CUB backend references CUB and Thrust version 1.16.0
+* `DeviceRadixSort` `num_items` parameter type is now templated instead of being an int
+  * If an integral type with a maximum size of 4 bytes is passed (an int), the former logic applies;
+    otherwise, the algorithm uses a larger indexing type that makes it possible to sort input data over
+    $2^32$ elements
```
Comment (saadrahim) [inline CHANGELOG.md]:
```suggestion
    $2^{32}$ elements
```

**Review by saadrahim — APPROVED** [review]

**Comment by saadrahim** [comment]
@LisaDelaney can you fix the merge conflicts?

**Review by umfranzw — APPROVED** [review]
This looks good to me.

**Pull request merged**